packages/Webkul/Payment/src/Payment.php

```diff
@@ -31,6 +32,10 @@ public function getPaymentMethods()
             $paymentMethod = app($paymentMethodConfig['class']);
 
             if ($paymentMethod->isAvailable()) {
+                if (! Cart::getCart()->haveStockableItems() && $paymentMethod->getCode() == 'cashondelivery') {
+                    continue;
+                }
```

Comment:
This is not the right place to add it.

Reply:
We can move it outside the outer if condition because it is reading an additional condition. If there is any other reason, could you please let me know?

Reply:
You can override the `isAvailable` method in CashOnDelivery class.

Reply:
The code has been placed in the appropriate place.